Jenkinsfile

```diff
@@ -1,14 +1,223 @@
+    }
+
     stages {
-        stage('no-op') {
+        stage('Initial setup') {
+            sh '''
+                mkdir -p $LABELS_DIR
+                mkdir -p $ARTIFACTS_DIR
+
+                # Save PR labels into a file
+                curl -s -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/project-koku/koku/issues/$ghprbPullId/labels | jq '.[].name' > $LABELS_DIR/github_labels.txt
+
+                cat $LABELS_DIR/github_labels.txt
+            '''
+        }
+
+        stage('Check PR check/smoke tests run') {
+            when {
+                expression {
+                    sh(script: "grep -E 'lgtm|pr-check-build|*smoke-tests|ok-to-skip-smokes' ${LABELS_DIR}/github_labels.txt")
+                }
+            }
+            steps {
+                sh '''
+                    
+
+                    task_arr=([1]="Build" [2]="Smoke Tests" [3]="Latest Commit")
+                    error_arr=([1]="The PR is not labeled to build the test image" [2]="The PR is not labeled to run smoke tests" [3]="This commit is out of date with the PR")
+
+                    if [ ! $(egrep 'lgtm|pr-check-build|*smoke-tests|ok-to-skip-smokes' ${LABELS_DIR}/github_labels.txt ]; then
+                        echo "PR check skipped; making skipped xml"
+
+                        cat << EOF > $WORKSPACE/artifacts/junit-pr_check.xml
+                        <?xml version="1.0" encoding="UTF-8" ?>
+                        <testsuite id="pr_check" name="PR Check" tests="1" failures="0">
+                            <testcase id="pr_check.skipped" name="Skipped">
+                            </testcase>
+                        </testsuite>
+                        EOF
+
+                    elif [ $(egrep 'ok-to-skip-smokes' ${LABELS_DIR}/github_labels.txt ]; then
+                        echo "smokes not required"
+
+                        cat << EOF > $WORKSPACE/artifacts/junit-pr_check.xml
+                        <?xml version="1.0" encoding="UTF-8" ?>
+                        <testsuite id="pr_check" name="PR Check" tests="1" failures="1">
+                            <testcase id="pr_check.${task_arr[$exit_code]}" name="${task_arr[$exit_code]}">
+                                <failure type="${task_arr[$exit_code]}">"${error_arr[$exit_code]}"</failure>
+                            </testcase>
+                        </testsuite>
+                        EOF
+
+                    exit 1
+                '''
+            }
+        }
+
+        stage('Build test image') {
+            when {
+                expression {
+                    sh(script: "egrep 'lgtm|pr-check-build|*smoke-tests' ${LABELS_DIR}/github_labels.txt || true", returnStdout: true) != true
+                }
+            }
+            steps {
+                withVault([configuration: configuration, vaultSecrets: secrets]) {
+                    sh '''
+                        if egrep 'aws-smoke-tests' ${LABELS_DIR}/github_labels.txt &>/dev/null
```

Comment:
This sets `IQE_FILTER_EXPRESSION` and optionally `IQE_MARKER_EXPRESSION` env variables, to affect the behavior of the `Run Smoke Tests` Stage later on, but I don't think the environment variables remain set between stages.